src/machine.ts

```diff
@@ -78,6 +79,65 @@ export const syncHandler = () => (callback, onReceive) => {
   };
 };
 
+// @ts-ignore
+export const mempoolHandler = () => (callback, onReceive) => {
+  logger.debug('Mempool handler instantiated');
+  const iterator = syncLatestMempool();
+  const asyncCall: () => void = async () => {
+    for (;;) {
+      const txResult: GeneratorYieldResult<MempoolEvent> = await iterator.next();
+      const { value, done } = txResult;
+
+      if (done) {
+        // The generator reached its end, we should end this handler
+        logger.debug('Done.', value)
+        break;
+      }
+
+      if (value && !value.success) {
+        logger.error(value.message);
+        // if there is any error calling the lambda, wait 30 seconds and retry
+        if (value.type == 'wait') {
+          callback('WAIT');
+          return;
+        }
+        callback('ERROR');
+        return;
+      }
+
+      if (value.type === 'finished') {
+        logger.info('Sync generator finished.');
+        callback('DONE');
+        return;
+      } else if (value.type === 'tx_success') {
+        logger.info('Mempool tx synced!');
+      } else {
+        logger.warn(`Unhandled type received from sync generator: ${value.type}`);
+      }
+    }
+
+    return;
+  };
+
+  onReceive((e: HandlerEvent) => {
+    switch(e.type) {
+      case 'START':
+        asyncCall();
+        break
+      case 'NEW_BLOCK':
```

Comment:
_Semi-related to line number:_

I think we should some tests to check if the machine is really pausing the mempool sync and transitioning to block sync if a `NEW_BLOCK` event is received and add a comment here explaining that we are pausing the mempool sync because the blocks sync has priority so we have more chance to have all transaction dependencies

Reply:
I tested this behavior on another simpler machine, it stops and  transitions to the block sync state.